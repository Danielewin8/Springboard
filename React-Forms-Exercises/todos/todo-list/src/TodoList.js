import React, { useState } from "react";
import { v4 as uuid } from 'uuid';
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
    // const INITIAL_STATE = [
    //     { id: uuid(), todo: "Clean room" }
    // ]
    const [todos, setTodos] = useState([])

    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, { ...newTodo }])
    }
    const deleteTodo = (e) => e.target.parentElement.remove();

    return (
        <div>
            <h1>Todos!</h1>
            <NewTodoForm addTodo={addTodo}/>
            <div className="todo-container">
                {todos.map(({ id, todo }) => 
                <Todo 
                    key={id}
                    todo={todo}
                    deleteTodo={deleteTodo}
                />
                )}
            </div>
        </div>
    )
}

export default TodoList;