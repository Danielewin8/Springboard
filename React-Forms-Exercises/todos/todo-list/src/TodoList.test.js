import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function () {
    render(<TodoList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it('can add a todo', () => {
    const { queryByText, getByLabelText } = render(<TodoList />);
    expect(queryByText('X')).not.toBeInTheDocument();
    expect(queryByText('Clean room')).not.toBeInTheDocument();
  
    const todo = getByLabelText("Add Todo");
    const button = queryByText('Add')
  
    fireEvent.change(todo, {target: {value: 'Clean Room'}});
    fireEvent.click(button);
  
    expect(queryByText('X')).toBeInTheDocument();
    expect(queryByText('Clean Room')).toBeInTheDocument();
})

it('can delete a todo', () => {
    const { queryByText, getByLabelText } = render(<TodoList />);
    expect(queryByText('X')).not.toBeInTheDocument();
    expect(queryByText('Clean room')).not.toBeInTheDocument();
  
    const todo = getByLabelText("Add Todo");
    const button = queryByText('Add')
  
    fireEvent.change(todo, {target: {value: 'Clean Room'}});
    fireEvent.click(button);
  
    expect(queryByText('X')).toBeInTheDocument();
    expect(queryByText('Clean Room')).toBeInTheDocument();

    const destroy = queryByText('X');
    fireEvent.click(destroy)

    expect(queryByText('X')).not.toBeInTheDocument();
    expect(queryByText('Clean room')).not.toBeInTheDocument();
})
