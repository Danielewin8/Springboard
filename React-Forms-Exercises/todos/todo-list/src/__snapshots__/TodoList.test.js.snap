// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Todos!
    </h1>
    <form>
      <label
        for="todo"
      >
        Add Todo
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Add Todo"
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
    <div
      class="todo-container"
    />
  </div>
</DocumentFragment>
`;
